odl/discr/lp_discr.py

```diff
@@ -449,15 +465,11 @@ def __getitem__(self, indices):
                     The resulting space with indexed domain and otherwise
                     same properties (except possibly weighting).
                 """
-                fspace = space.uspace.byaxis_in[indices]
+                fspace = space.fspace.byaxis_in[indices]
                 part = space.partition.byaxis[indices]
 
-                if isinstance(space.weighting, NoWeighting):
```

Comment:
Where did this go? Overall NoWeighting is different from ConstWeighting (conceptually at least), For example, if we make a ray transform with domain `odl.uniform_discr([-1, -1], [1, 1], [2, 2])`, we still want the range to be weighted, even though the domain has weighting 1 and is thus in some sense equivalent to NoWeighting.

Reply:
I don't see how the weightings of domain and range are connected for ray transforms. They're totally different spaces.

The only situation where weighting vs. no weighting makes a difference is when the cell sizes are changed from 1 to something else, or the other way around; plus the case when axes are removed or added.
The only place where we change cell sizes is `uniform_discr_fromdiscr`, I'll have to check what we do with weightings there.
Removing axes will necessarily destroy weightings, unless there's a way to regenerate the lower-dimensional versions of them. This possibility exists in `DiscreteLp`, but not in `TensorSpace`. Later on, when we implement per-axis weighting, that will be preserved, too.